### PR TITLE
Add support for Graphite web sigmoid() function

### PR DESCRIPTION
**Pull request by carrieedwards**
This PR adds support for the Graphite web sigmoid() function.

The sigmoid function is defined as:

```
sigmoid(seriesList)
Takes one metric or a wildcard seriesList and applies the sigmoid function 1 / (1 + exp(-x)) to each datapoint.

Example:

&target=sigmoid(Server.instance01.threads.busy)
&target=sigmoid(Server.instance*.threads.busy)
```

### CODE REVIEW

File: expr/functions/sigmoid/function_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+func TestFunction(t *testing.T) {
+	now32 := int64(time.Now().Unix())
+
+	tests := []th.EvalTestItem{
+		{
+			"sigmoid(metric1)",
+			map[parser.MetricRequest][]*types.MetricData{
+				{"metric1", 0, 1}: {types.MakeMetricData("metric1", []float64{5, 1, math.NaN(), 0, 12, 125, 10.4, 1.1}, 60, now32)},
```
Comment (ywwg) [inline expr/functions/sigmoid/function_test.go:32]: can you test the -1 situation since we know that's a special case?

**Review by ywwg — APPROVED** [review]
just need to cover the edge cases explicitly

File: expr/functions/sigmoid/function_test.go (hunk 1 above)
Comment (carrieedwards) [reply expr/functions/sigmoid/function_test.go:32]: Actually, is there an e^(-x) value that would result in -1? since e^0 is 1

Comment (alexgreenbank) [reply expr/functions/sigmoid/function_test.go:32]: Euler's Identity is: e^(iπ) = -1 but I really don't think you need to worry about complex numbers.

No non-complex exponent can give a value of e^x < 0.

For e^x:
* As x --> infinity e^x --> infinity
* At x=0 e^x = 1
* As x --> -infinity e^x --> 0
